File: blogtruyen/src/news.lua
local url, page = ...
if text:is_empty(page) then
    page = 1
end

local doc = http:get(url ..  "-"  .. page):html()

if doc ~= nil then
    local el = doc:select(".list-mainpage .storyitem")
    local novelList = {}
    local next

    local last = doc:select(".pagination"):select("li:has(.glyphicon-step-forward)"):last():select("a")
    if last ~= nil then
        next = regexp:find(last, "-(\\d+)")
    end
    for i = 1, el:size() do
        local e = el:get(i - 1)
        local novel = {
            ["name"] = e:select(".title a"):attr("title"),
            ["link"] = e:select(".title a"):first():attr("href"),
            ["description"] = e:select(".statictis"):last():text(),
            ["cover"] = e:select("img"):first():attr("src"),
            ["host"] = "https://blogtruyen.vn"
        }
        table.insert(novelList, novel)
    end

    return response:success(novelList, next)
end

return nil

File: blogtruyen/src/news.js
function execute(url, page) {
    if (!page) page = '1';

    let response = fetch(url + "-" + page);

    if (response.ok) {
        let doc = response.html();

        let next = doc.select(".pagination").select("li:has(.glyphicon-step-forward)").last().select("a").attr("href").match(/-(\d+)/);
        if (next) next = next[1]; else next = '';
        let novelList = [];
        doc.select(".list-mainpage .storyitem").forEach(e => novelList.push({
            name: e.select(".title a").attr("title"),
            link: e.select(".title a").first().attr("href"),
            description: e.select(".statictis").last().text(),
            cover: e.select("img").first().attr("src"),
            host: "https://blogtruyen.vn"
        }));
        return Response.success(novelList, next)
    }
    return null;
}